src/Config/XmlConfigFileLoader.php

```diff
@@ -0,0 +1,220 @@
+     */
+    public function __construct(ConfigBuilder $configBuilder, string $applicationRootDirectory)
+    {
+        $this->configBuilder = $configBuilder;
+        $this->applicationRootDirectory = $applicationRootDirectory;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function isSupported(string $file): bool
+    {
+        return in_array(pathinfo($file, PATHINFO_BASENAME), ['.psh.xml', '.psh.xml.dist', '.psh.xml.override'], true);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function load(string $file, array $params): Config
+    {
+        $pshConfigNode = $this->loadXmlRoot($file);
+        $this->configBuilder->start();
+
+        try {
+            $this->configBuilder->setHeader(
+                $this->extractNode(self::NODE_HEADER, $pshConfigNode)->nodeValue
+            );
+        } catch (\InvalidArgumentException $e) {
+        }
+
+        $this->setConfigData($file, $pshConfigNode);
+
+        $environments = $this->extractNodes(self::NODE_ENVIRONMENT, $pshConfigNode);
+
+        foreach ($environments as $node) {
+            $this->configBuilder->start($node->getAttribute('name'));
+            $this->setConfigData($file, $node);
+        }
+
+        return $this->configBuilder
+            ->create($params);
+    }
+
+    /**
+     * @param string $file
+     * @param array $pshConfigNode
+     */
+    private function setConfigData(string $file, DOMElement $pshConfigNode)
+    {
+        $this->configBuilder->setCommandPaths(
+            $this->extractCommandPaths($file, $pshConfigNode)
+        );
+
+        try {
+            $this->extractPlaceholders(
+                $this->extractNode(self::NODE_PLACEHOLDER, $pshConfigNode)
+            );
+        } catch (\InvalidArgumentException $e) {
+            // nth
+        }
+
+        $this->configBuilder->setTemplates(
+            $this->extractTemplates($file, $pshConfigNode)
+        );
+    }
+
+    /**
+     * @param string $key
+     * @param DOMElement $parent
+     * @return DOMElement[]
+     */
+    private function extractNodes(string $key, DOMElement $parent): array
+    {
+        $nodes = [];
+
+        foreach ($parent->childNodes as $childNode) {
+            if ($childNode instanceof DOMElement && $childNode->localName === $key) {
+                $nodes[] = $childNode;
+            }
+        }
+
+        if (0 === count($nodes)) {
+            return [];
+        }
+
+        return $nodes;
+    }
+
+    /**
+     * @param string $key
+     * @param DOMElement $parent
+     * @return DOMElement
+     */
+    private function extractNode(string $key, DOMElement $parent): DOMElement
+    {
+        $nodes = $this->extractNodes($key, $parent);
+
+        if (count($nodes) === 0) {
+            throw new \InvalidArgumentException('No node found for ' . $key);
+        }
+
+        return $nodes[0];
+    }
+
+    /**
+     * @param string $file
+     * @param $pshConfigNode
+     * @return array
+     */
```

Comment:
Der php doc block kann von mir aus weg. Steht ja alles schon in der Methoden Signatur drin :-)